mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -815,4 +829,14 @@ public boolean hasChildren(MailboxPath mailboxPath, MailboxSession session) thro
         Mailbox mailbox = mapper.findMailboxByPath(mailboxPath);
         return mapper.hasChildren(mailbox, session.getPathDelimiter());
     }
+
+    @Override
+    public void setLimitOfAnnotations(int limitOfAnnotations) {
```

Comment:
There's a lot of questions here.
First, about the number of parameters :
* Having too many parameters usually shows that the class does too much : that's clearly the case, but do you want to break it into small parts right now ?
* It's often painful because you don't want to call the constructor, here it's not a problem because of injection, but we would use a builder in another case

Second, the lifecycle of objects is really important : if you allow passing a configuration variable into a setter, does that mean that you can change at anytime ? Usually, we don't want to deal with this kind of complexity, and showing that the parameter is immutable and passed at construction time is so valuable that we prefer handling a refactoring than adding setters to a class.

To conclude about that : there are already setters in this class, so harm is already done. You can skip the refactoring about that but you should at least make it clear that this method should be called before `init()` and handle is value like other setters do.